aten/src/ATen/core/type.cpp

```diff
@@ -1410,14 +1410,48 @@ VaryingShape<Stride> TensorType::computeStrideProps(
   std::vector<size_t> stride_indices(sizes.size());
   std::iota(stride_indices.begin(), stride_indices.end(), 0);
 
+  // Sorting strides in ascending order
+  // Warning: A tensor that has more than one dimension of size 1 has
+  // insufficient information to recreate the contiguous order of its indices.
+  // Ties are broken based on whether one of the dimensions is of size
+  // one.  When two dimensions have the same stride, the stride
+  // associated with a dimension of size 1 is considered "smaller"
+  // as it created the condition for the second stride of the same size.
+  // Example:
+  //  Prior to sorting
+  //  Idx:     [0,   1,  2,  3]
+  //  sizes:   [8,   1, 10, 16]
+  //  Strides: [160, 1, 16,  1]
+  //  After sorting
+  //  Idx:     [1,  3,  2,   0]
+  //  sizes:   [1, 16, 10,   8]
+  //  Strides: [1,  1, 16, 160]
   std::sort(
       stride_indices.begin(),
       stride_indices.end(),
-      [&strides](const int& a, const int& b) {
-        // break ties in case of unsqueezed dims
-        // i.e. (1, 1, 5)
+      [&strides, &sizes](const int& a, const int& b) {
         if (strides[a] == strides[b]) {
-          return a > b;
+          // when strides of two axes are identical:
+          // i. if it's not self-overlapping:
+          //   1. if axis_a is the only size-1 axis, we think axis_a comes
+          //      before axis_b. Since strides[b] == strides[a] * sizes[a];
+          //      Meanwhile, it is not true for axis a;
+          //   2. if axis_b is the only size-1 axis, likewise, we think axis_a
```

Comment:
nit: if we are using `likewise` could we say "we think axis_b comes before axis_a?